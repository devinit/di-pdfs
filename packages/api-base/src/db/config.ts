import {TConfig} from 'pg-promise';

// get the connection details from Allan or Donata

export const dwConfig: TConfig = {
    user: process.env.DB_USER,
    database: process.env.DB_DATABASE,
    host: process.env.DB_HOST,
    port: Number(process.env.DB_PORT),
    password: process.env.DB_PASSWORD
};
