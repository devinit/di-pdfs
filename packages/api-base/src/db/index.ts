import {IMain, IDatabase, IOptions} from 'pg-promise';
import {dwConfig} from './config';
import * as pgPromise from 'pg-promise';
import * as LRU from 'lru-cache';

export interface IExtensions {
    manyCacheable: (query: string, values?: object) => Promise<any[]>;
}

const lruOpts: LRU.Options<any> = {
    max: 300,
    maxAge: 1000 * 60 * 60 * 60
};

export const dbCache: LRU.Cache<any, any> = LRU(lruOpts);

// pg-promise initialization options:
const options: IOptions<IExtensions> = {
    // Extending the database protocol with our custom modules
    // API: http://vitaly-t.github.io/pg-promise/global.html#event:extend
    extend: (obj: IExtensions & IDatabase<IExtensions>) => {
        obj.manyCacheable = (query: string, values?: any) => {
            const getQuery = values ? pgPromise.as.format(query, values) : query;
            if (dbCache.has(getQuery)) {
                return Promise.resolve(dbCache.get(getQuery));
            }
            return obj.any(getQuery);
        };
    },
    // caching
    receive: (data, _result, event) => {
        // cache recieved
        if (!dbCache.has(event.query)) dbCache.set(event.query, data);
    }
};

const pgp: IMain = pgPromise(options);

// Create the database instance with extensions:
const db = pgp<IExtensions>(dwConfig);

process.on('exit', (code) => {
  // kill db
  pgp.end();
  console.log(`About to exit with code: ${code}, closed DB connection`);
  process.kill(process.pid);
});

process.on('SIGINT', () => {
    // kill db
    pgp.end();
    console.log('Ctrl-C... forced termination closed DB connection');
    process.kill(process.pid);
});

export type IDB = IDatabase<IExtensions> & IExtensions; // necessary repition to make typescript happy
// If you ever need access to the library's root (pgp object), you can do it via db.$config.pgp
// See: http://vitaly-t.github.io/pg-promise/Database.html#.$config
export default db;
