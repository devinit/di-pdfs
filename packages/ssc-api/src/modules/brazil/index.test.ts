import * as prettyFormat from 'pretty-format';
import {Brazil} from '.';
import db from '@di-pdfs/api-base/lib/db';

describe('Brazil', () => {
    const brazil = new Brazil(db);
    it('getting dvptCooperation', async () => {
       const data = await brazil.dvptCooperation();
       expect(prettyFormat(data)).toMatchSnapshot();
    }, 100000);
    it('getting tch cooperation', async () => {
        const data = await brazil.tchCooperationByRegion();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    it('getting tch top recipients', async () => {
        const data = await brazil.topTchRecipients();
        expect(prettyFormat(data)).toMatchSnapshot();
     }, 100000);
    afterAll(() => {
       db.$config.pgp.end();
    });
});
