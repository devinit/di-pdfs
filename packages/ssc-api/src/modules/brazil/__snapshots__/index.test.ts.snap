// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Brazil getting tch cooperation 1`] = `
"Array [
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 43,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 48.9,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"region\\": \\"Oceania\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 4.3,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"region\\": \\"Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 2.3,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#FECB2F\\",
    \\"region\\": \\"Asia and the Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"North America\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.5,
    \\"year\\": 2011,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 44.3,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 46.9,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"region\\": \\"Oceania\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3.9,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"region\\": \\"Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 2.6,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#FECB2F\\",
    \\"region\\": \\"Asia and the Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 1.7,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"North America\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.6,
    \\"year\\": 2012,
  },
  Object {
    \\"color\\": \\"#e8443a\\",
    \\"region\\": \\"Africa\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 53.3,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"Latin America and the Caribbean\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 39.4,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#EA7600\\",
    \\"region\\": \\"Oceania\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 2.9,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#93328E\\",
    \\"region\\": \\"Europe\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 3.3,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#FECB2F\\",
    \\"region\\": \\"Asia and the Middle East\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.9,
    \\"year\\": 2013,
  },
  Object {
    \\"color\\": \\"#0095CB\\",
    \\"region\\": \\"North America\\",
    \\"uid\\": \\"unique_id_stub\\",
    \\"value\\": 0.3,
    \\"year\\": 2013,
  },
]"
`;
